src/Umbraco.Infrastructure/Compose/RelateOnTrashComponent.cs

```diff
@@ -68,47 +76,55 @@ private void MediaService_Moved(IMediaService sender, MoveEventArgs<IMedia> e)
-                //may not be the case if the encompassing transaction also deleted it when this item was moved to the bin
+                    relationType = new RelationType(relationTypeName, relationTypeAlias, false, documentObjectType,
+                        documentObjectType);
+                    _relationService.Save(relationType);
+                }
 
-                if (_entityService.Exists(originalParentId))
+                foreach (var item in e.MoveInfoCollection)
                 {
-                    // Add a relation for the item being deleted, so that we can know the original parent for if we need to restore later
-                    var relation = new Relation(originalParentId, item.Entity.Id, relationType);
-                    _relationService.Save(relation);
-
-                    _auditService.Add(AuditType.Delete,
-                        item.Entity.WriterId,
-                        item.Entity.Id,
-                        ObjectTypes.GetName(UmbracoObjectTypes.Document),
-                        string.Format(_textService.Localize(
-                                "recycleBin/contentTrashed"),
-                            item.Entity.Id, originalParentId));
+                    var originalPath = item.OriginalPath.ToDelimitedList();
+                    var originalParentId = originalPath.Count > 2
+                        ? int.Parse(originalPath[originalPath.Count - 2])
+                        : Constants.System.Root;
+
+                    //before we can create this relation, we need to ensure that the original parent still exists which
+                    //may not be the case if the encompassing transaction also deleted it when this item was moved to the bin
+
+                    if (_entityService.Exists(originalParentId))
+                    {
+                        // Add a relation for the item being deleted, so that we can know the original parent for if we need to restore later
+                        var relation = _relationService.GetByParentChildAndType(originalParentId, item.Entity.Id, relationType) ??
+                                       new Relation(originalParentId, item.Entity.Id, relationType);
+                        _relationService.Save(relation);
+
+                        _auditService.Add(AuditType.Delete,
+                            item.Entity.WriterId,
+                            item.Entity.Id,
+                            ObjectTypes.GetName(UmbracoObjectTypes.Document),
+                            string.Format(_textService.Localize(
+                                    "recycleBin/contentTrashed"),
+                                item.Entity.Id, originalParentId));
+                    }
                 }
+
+                scope.Complete();
             }
         }
 
-        private void MediaService_Trashed(IMediaService sender, MoveEventArgs<IMedia> e)
+        public void MediaService_Trashed(IMediaService sender, MoveEventArgs<IMedia> e)
```

Comment:
why public?

Reply:
Just revert, my bad.. This was justed tempory to test I found have the same error before I moved the tests